CHANGELOG.md

```diff
@@ -15,6 +15,14 @@
 [All changes in [[UnreleasedUniFFIVersion]]](https://github.com/mozilla/uniffi-rs/compare/v0.24.1...HEAD).
 
 ### What's new
+- Libraries which use UDL should start using `uniffi::gen_scaffolding(udl_path, crate_name)` to generate
+  Rust scaffolding - the new argument is the name of the crate library (the name under the `[lib]` entry
+  in your `Cargo.toml` - ie, the name other Rust code would use to reference your types).
+  The old function, `uniffi::generate_scaffolding(udl_path)` is soft-deprecated because
+  not knowing the crate name causes interoperability issues sharing types between UDL and proc-macros
+  when the namespace in the UDL doesn't match the crate name.
+  We expect it will be hard-deprecated (ie, issue warnings) in a few versions and fully removed in
+  an even later version.
```

Comment:
I'd love to avoid adding this extra pain for our users.  What about the process we discussed last week where `generate_scaffolding` generates proc-macro calls, then the proc-macros generate the actual scaffolding code with the correct crate name?

Alternatively, maybe we could soften the language up even more.  Rather than say `generate_scaffolding` is soft-deprecated, we could say that `gen_scaffolding` is a workaround for interoperability issues as we investigate ways to implement a system that doesn't need it.